### PR TITLE
feat: move next hearing date field to after CMO approval (FPLA-1107)

### PR DESCRIPTION
**Pull request by Jamesnelson117**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-1107

### Change description ###
Next hearing Date (Action CMO) now appears after Judge select 'Send to all parties' in the order action page. 
Next hearing details only appears on issued CMO (PDF)

Screenshot 2019-12-16 at 16 12 16

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 import static org.apache.commons.lang3.StringUtils.defaultIfBlank;
 import static uk.gov.hmcts.reform.fpl.service.CaseDataExtractionService.EMPTY_PLACEHOLDER;
+import static uk.gov.hmcts.reform.fpl.service.CaseDataExtractionService.HEARING_EMPTY_PLACEHOLDER;
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java]: I would suggest moving `HEARING_EMPTY_PLACEHOLDER` to either this class or `DocmosisTemplateDataGeneration`, think those are more logical places for it

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDataGeneratorHelper.java [hunk 2]
```diff
@@ -540,4 +538,15 @@ public static DynamicList createHearingBookingDynmaicList() {
             .listItems(List.of(DynamicListElement.builder().code(UUID.randomUUID()).label("test").build()))
             .build();
     }
+
+    public static CaseManagementOrder createApprovedCMO() {
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDataGeneratorHelper.java]: Sam would be proud of you not using `CaseDataGeneratorHelper`

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseManagementOrderServiceTest.java [hunk 3]
```diff
@@ -27,16 +27,13 @@
...
 import static uk.gov.hmcts.reform.fpl.utils.DocumentManagementStoreLoader.document;
 
 class CaseManagementOrderServiceTest {
     private static final LocalDateTime NOW = LocalDateTime.now();
     private final Time time = () -> NOW;
-    private static final LocalDateTime DATE = LocalDateTime.of(2018, 2, 12, 9, 30);
 
-    private CaseManagementOrderService service = new CaseManagementOrderService(time,
-        new DateFormatterService(), new HearingBookingService());
+    private CaseManagementOrderService service = new CaseManagementOrderService(time, new HearingBookingService());
```
Comment (lewisbirks) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CaseManagementOrderServiceTest.java]: Still not sure why we haven't autowired in an instance of time, would mean that you don't need to manually create CaseManagementOrderService

File: service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDataGeneratorHelper.java (hunk 2 above)
Comment (swalker125) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/utils/CaseDataGeneratorHelper.java]: 😠

**Pull request merged**